f5_openstack_agent/lbaasv2/drivers/bigip/irule.py

```diff
@@ -0,0 +1,234 @@
+# coding=utf-8
+# Copyright (c) 2016-2018, F5 Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from f5_openstack_agent.lbaasv2.drivers.bigip import resource_helper
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class iRuleHelper(object):
+    """A tool class for tcp irule process"""
+
+    def __init__(self):
+        self.irule_helper = resource_helper.BigIPResourceHelper(
+            resource_helper.ResourceType.rule
+        )
+        self.vs_helper = resource_helper.BigIPResourceHelper(
+            resource_helper.ResourceType.virtual
+        )
+
+        self.delete_iRule = False
+
+    @staticmethod
+    def create_v4_iRule_content(tcp_options):
+        template = """when SERVER_INIT {
+            scan [getfield [IP::client_addr] "%%" 1] {%%d.%%d.%%d.%%d} a b c d
```

Comment:
Yes, It may be possible. we need to confirm this with the SEs. If we allow an editable rule outside the code, it might be another feature?

Reply:
> Is it possible that customer or SEs inform us to change the content frequently. Is it necessary to put them into a separate configuration file, just like `f5-extended-profile.json`?

Yes, It may be possible. we need to confirm this with the SEs. If we allow an editable rule outside the code, it might be another feature?

Reply:
> > Is it possible that customer or SEs inform us to change the content frequently. Is it necessary to put them into a separate configuration file, just like `f5-extended-profile.json`?
> 
> Yes, It may be possible. we need to confirm this with the SEs. If we allow an editable rule outside the code, it might be another feature?

It's ok to put it as another feature in the future. 
I just don't want to change the code every time according to requests from SE.